packages/base-controller/src/ControllerMessenger.test.ts

```diff
@@ -271,8 +271,11 @@ describe('ControllerMessenger', () => {
         getPayload: () => [state],
       });
       const handler = sinon.stub();
-      const selector = sinon.fake((obj: Record<string, unknown>) => obj.propA);
-      controllerMessenger.subscribe('complexMessage', handler, selector);
+      controllerMessenger.subscribe(
+        'complexMessage',
+        handler,
+        (obj) => obj.propA,
```

Comment:
Before this change, this argument would have needed a type, and now this is properly inferred.

Reply:
f281d49f4c73a3da4bd6338faed1f367a04c9bd1 was exactly what I was thinking of. 

The "should allow publishing multiple different events to subscriber" test case specifically verifies that with these changes, the `EventType` generic param of the `ControllerMessenger.subsribe` method enables it to narrow its selector function from a union of multiple events down to the selected event.

One think we might need to add is a breaking changelog entry specifying that the generic API of `SelectorFunction` has changed (number and order of arguments, type constraint for `EventType`).

Otherwise LGTM!